src/example/test_border.py

```diff
@@ -0,0 +1,65 @@
+
+    graph = Graph()
+    nodes = []
+    edges = []
+    nodes.append(Node(NodeAttrs('v', 0 , 0 , flag=False)))
+    nodes.append(Node(NodeAttrs('v', 1 , 0 , flag=False)))
+    nodes.append(Node(NodeAttrs('v', 1 , 1 , flag=False)))
+    nodes.append(Node(NodeAttrs('v', 0 , 1 , flag=False)))
+    nodes.append(Node(NodeAttrs('v',0.5,0.5, flag=False)))
+    nodes.append(Node(NodeAttrs('v',0.5, 0 , flag=False)))
+    nodes.append(Node(NodeAttrs('v', 1 ,0.5, flag=False)))
+    nodes.append(Node(NodeAttrs('v',0.5, 1 , flag=False)))
+    nodes.append(Node(NodeAttrs('v', 0 ,0.5, flag=False)))
+    nodes.append(Node(NodeAttrs('q',0.25,0.25,flag=True)))
+    nodes.append(Node(NodeAttrs('q',0.75,0.25,flag=False)))
+    nodes.append(Node(NodeAttrs('q',0.75,0.75,flag=False)))
+    nodes.append(Node(NodeAttrs('q',0.25,0.75,flag=False)))
+    
+    edges.append(Edge(0,5,EdgeAttrs(kind='e', flag=True, handle=16)))
+    edges.append(Edge(0,8,EdgeAttrs(kind='e', flag=True, handle=23)))
+    edges.append(Edge(0,9,EdgeAttrs(kind='q', flag=False, handle=32)))
+    edges.append(Edge(1,5,EdgeAttrs(kind='e', flag=True, handle=17)))
+    edges.append(Edge(1,6,EdgeAttrs(kind='e', flag=True, handle=18)))
+    edges.append(Edge(1,10,EdgeAttrs(kind='q', flag=False, handle=33)))
+    edges.append(Edge(2,6,EdgeAttrs(kind='e', flag=True, handle=19)))
+    edges.append(Edge(2,7,EdgeAttrs(kind='e', flag=True, handle=20)))
+    edges.append(Edge(2,11,EdgeAttrs(kind='q', flag=False, handle=34)))
+    edges.append(Edge(3,7,EdgeAttrs(kind='e', flag=True, handle=21)))
+    edges.append(Edge(3,8,EdgeAttrs(kind='e', flag=True, handle=22)))
+    edges.append(Edge(3,12,EdgeAttrs(kind='q', flag=False, handle=35)))
+    edges.append(Edge(5,4,EdgeAttrs(kind='e', flag=False, handle=24)))
+    edges.append(Edge(5,9,EdgeAttrs(kind='q', flag=False, handle=28)))
+    edges.append(Edge(5,10,EdgeAttrs(kind='q', flag=False, handle=29)))
+    edges.append(Edge(6,4,EdgeAttrs(kind='e', flag=False, handle=25)))
+    edges.append(Edge(6,10,EdgeAttrs(kind='q', flag=False, handle=29)))
+    edges.append(Edge(6,11,EdgeAttrs(kind='q', flag=False, handle=30)))
+    edges.append(Edge(7,4,EdgeAttrs(kind='e', flag=False, handle=26)))
+    edges.append(Edge(7,11,EdgeAttrs(kind='q', flag=False, handle=30)))
+    edges.append(Edge(7,12,EdgeAttrs(kind='q', flag=False, handle=31)))
+    edges.append(Edge(8,4,EdgeAttrs(kind='e', flag=False, handle=27)))
+    edges.append(Edge(8,9,EdgeAttrs(kind='q', flag=False, handle=28)))
+    edges.append(Edge(8,12,EdgeAttrs(kind='q', flag=False, handle=31)))
+    edges.append(Edge(4,9,EdgeAttrs(kind='q', flag=False, handle=36)))
+    edges.append(Edge(4,10,EdgeAttrs(kind='q', flag=False, handle=37)))
+    edges.append(Edge(4,11,EdgeAttrs(kind='q', flag=False, handle=38)))
+    edges.append(Edge(4,12,EdgeAttrs(kind='q', flag=False, handle=39)))
+    graph.add_node_collection(nodes)
+    graph.add_edge_collection(edges)
+
+    graph.display()
+    plt.show()
+
+    P1()(graph)
+
```

Comment:
What is this file for?

Reply:
This was a test for visualizing the problem with edge_equality. Now it's basically useless.